File: chakeng/src/main/java/com/ssafy/chakeng/common/ApiResponse.java
package com.ssafy.chakeng.common;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ApiResponse<T> {
    private boolean success;
    private String code;
    private String message;
    private T data;

    public static <T> ApiResponse<T> ok(String message, T data) {
        return new ApiResponse<>(true, "OK", message, data);
    }

    public static <T> ApiResponse<T> error(String code, String message) {
        return new ApiResponse<>(false, code, message, null);
    }
}

File: chakhaeng-android/app/src/main/java/com/sos/chakhaeng/data/network/dto/ApiResponse.kt
package com.sos.chakhaeng.data.network.dto

data class ApiResponse<T>(
    val success: Boolean,
    val code: String,
    val message: String,
    val data: T?
)

// 디버깅을 위한 장치
inline fun <T> ApiResponse<T>.getOrThrow(): T {
    if (success && data != null) return data
    throw IllegalArgumentException("${code ?: ""} ${message ?: ""}".trim())
}